### PR TITLE
Updating pattern X and artifacts for RC1

### PR DESCRIPTION
**Pull request by RakhithaRR**

### CODE REVIEW

File: artifacts/install/api-manager-artifacts/pattern-1/wso2am-p1-analytics-dash-conf.yaml [hunk 1]
```diff
@@ -1,6 +1,405 @@
 apiVersion: v1
+# data:
```
Comment (pubudu538) [inline artifacts/install/api-manager-artifacts/pattern-1/wso2am-p1-analytics-dash-conf.yaml]: Please remove the commented data.

File: pkg/apis/apim/v1alpha1/types.go [hunk 2]
```diff
@@ -158,6 +158,7 @@ type APIManagerSpec struct {
 	Replicas       *int32             `json:"replicas"`
 	Profiles 	[]Profile  				`json:"profiles"`
 	Service Service `json:"service"`
+	UseVolume bool `json:"useVolume"`
```
Comment (pubudu538) [inline pkg/apis/apim/v1alpha1/types.go]: There is no need to introduce a new flag. When processing the custom pattern, you just need to ignore those volumes. Shall we do that change?

**Review by pubudu538 — CHANGES_REQUESTED** [review]

File: artifacts/install/api-manager-artifacts/pattern-1/wso2am-p1-analytics-dash-conf.yaml (hunk 1 above)
Comment (RakhithaRR) [reply artifacts/install/api-manager-artifacts/pattern-1/wso2am-p1-analytics-dash-conf.yaml]: done

File: pkg/apis/apim/v1alpha1/types.go (hunk 2 above)
Comment (RakhithaRR) [reply pkg/apis/apim/v1alpha1/types.go]: I'll look into this.

**Review by pubudu538 — APPROVED** [review]

File: artifacts/install/operator-configs/controller-conf.yaml [hunk 3]
```diff
@@ -48,30 +48,30 @@ data:
...
-  p1-apim-deployment-image: "wso2/wso2am:3.0.0"
+  p1-apim-deployment-image: "pubudu/wso2am:3.1.0-rc1"
   p1-apim-deployment-resources-requests-cpu: "2000m"
   p1-apim-deployment-resources-requests-memory: "2Gi"
   p1-apim-deployment-resources-limits-cpu: "2000m"
   p1-apim-deployment-resources-limits-memory: "2Gi"
 
   # custom pattern specific configs for apim-instances
-  pX-apim-deployment-image: "wso2/wso2am:3.0.0"
+  pX-apim-deployment-image: "pubudu/wso2am:3.1.0-rc1"
```
Comment (menakaj) [inline artifacts/install/operator-configs/controller-conf.yaml:58]: Why there are user-specific image details?

File: pkg/controller/controller.go [hunk 4]
```diff
@@ -1072,114 +1085,115 @@ func (c *Controller) syncHandler(key string) error {
 
 		}
 
+		if apimVolDefined && dashVolDefined && workerVolDefined {
```
Comment (menakaj) [inline pkg/controller/controller.go:1090]: Shall we remove the additional lines if this file is not an auto-generated one?

File: pkg/controller/pattern1/deployment.go [hunk 5]
```diff
@@ -433,7 +433,8 @@ func DashboardDeployment(apimanager *apimv1alpha1.APIManager,y *configvalues, nu
 					Containers: []corev1.Container{
 						{
 							Name:  "wso2am-pattern-1-analytics-dashboard",
-							Image: "wso2/wso2am-analytics-dashboard:3.0.0",
+							// Image: "wso2/wso2am-analytics-dashboard:3.0.0",
+							Image: "pubudu/wso2am-analytics-dashboard:3.1.0-rc1",
```
Comment (menakaj) [inline pkg/controller/pattern1/deployment.go:437]: Image location shall we use the WSO2 repository.

File: pkg/apis/apim/v1alpha1/types.go (hunk 2 above)
Comment (RakhithaRR) [reply pkg/apis/apim/v1alpha1/types.go]: Fixed

File: artifacts/install/operator-configs/controller-conf.yaml (hunk 3 above)
Comment (RakhithaRR) [reply artifacts/install/operator-configs/controller-conf.yaml:58]: Used only for pre-release testing. Fixed in latest code.

File: pkg/controller/pattern1/deployment.go (hunk 5 above)
Comment (RakhithaRR) [reply pkg/controller/pattern1/deployment.go:437]: Used only for pre-release testing. Fixed in latest code.

File: pkg/controller/controller.go (hunk 4 above)
Comment (RakhithaRR) [reply pkg/controller/controller.go:1090]: These have been removed in the latest code.

**Pull request merged**